dask/array/routines.py

```diff
@@ -639,29 +645,42 @@ def bincount(x, weights=None, minlength=0, split_every=None):
         if weights.chunks != x.chunks:
             raise ValueError("Chunks of input array x and weights must match.")
 
+    axis = (0,)
     token = tokenize(x, weights, minlength)
     args = [x, "i"]
     if weights is not None:
-        meta = np.bincount([1], weights=[1])
+        meta = array_safe(np.bincount([1], weights=[1]), x._meta)
```

Comment:
I'm not sure if there's any case where `x` could be a non-Dask array, but if that happens this will fail when trying to access `x._meta`. Could we use `like=meta_from_array(x)` here? This would be similar to the following example: https://github.com/dask/dask/blob/82fcaab087621c750dc41b453e9681f81f753abe/dask/array/routines.py#L673